tools/lsp/preview/element_selection.rs

```diff
@@ -456,7 +456,18 @@ fn filter_nodes_for_selection(
         return None;
     }
 
-    selection_candidate.as_element_node()
+    let en = selection_candidate.as_element_node();
+
+    if let Some(en) = &en {
+        if en
+            .with_element_node(|n| n.parent().map(|p| p.kind() == SyntaxKind::Component))
+            .unwrap_or_default()
+        {
+            return None;
+        }
+    }
+
+    en
```

Comment:
Suggestion for shortening, but feel free to ignore :)

```suggestion
    selection_candidate.as_element_node()
        .filter(|en| en.with_element_node(|n| n.parent().map_or(true, |p| p.kind() != SyntaxKind::Component)))
```